route/v1/types.go

```diff
@@ -369,14 +369,17 @@ type RouteIngressConditionType string
 const (
 	// RouteAdmitted means the route is able to service requests for the provided Host
 	RouteAdmitted RouteIngressConditionType = "Admitted"
-	// TODO: add other route condition types
+	// RouteDeprecated indicates that the route is using a deprecated configuration that may
+	// be incompatible with a future version of OpenShift. If this set to true on any route,
+	// the ingress operator may block upgrades.
```

Comment:
micro-nit, feel free to ignore.

> If this set to true on any route, the ingress operator may block upgrades.

This seems like an internal implementation detail.  In cases where the ingress operator decides to use `Deprecated` on `Routes` to drive `Upgradeable=False`, that `Upgradeable` ClusterOperator condition is the API talking to the CVO, and the CVO will in turn set `Upgradeable=False` on ClusterVersion, and that ClusterVersion condition is the API talking to the cluster-admin or other OCP-managing actor.  I don't think we need to hint at that possible chain in this Route condition's docs.  Instead, we can keep the first sentence explaining the deprecation aspect, which lets Route-admins know that they should be thinking about a config update.  And then leave it to the cluster-admin-facing ClusterVersion API to talk about whether we're concerned about anything (including deprecated Route config) to talk to cluster-admins about that aspect.

But also, "may block upgrades" is wiggly enough that I doubt anyone will feel like we are surprising them regardless of whether the ingress operator decides to take any action based on this new Route condition.  So no worries if you want to leave this sentence in as a semi-informative bread-crumb.

Reply:
thanks for the info - I remove it as I agree it's an implementation detail.

Reply:
Whoops, I forgot to remove this.